packages/angular/cli/models/parser.ts

```diff
@@ -259,6 +259,8 @@ export function parseFreeFormArguments(args: string[]): Arguments {
  * @returns An object that contains a property per option.
  */
 export function parseArguments(args: string[], options: Option[] | null): Arguments {
+  args = args.slice();
```

Comment:
```suggestion
export function parseArguments(originArgs: string[], options: Option[] | null): Arguments {
  const args = [ ...originArgs ];
```
Something like this?

Reply:
I'm a little bit of confused, wouldn't the below tweaks guarantees the `args` been local from the beginning? (already renamed the param form `args` to `originArgs`)

```diff
- export function parseArguments(args: string[], options: Option[] | null): Arguments {
+ export function parseArguments(originArgs: string[], options: Option[] | null): Arguments {
+  const args = [ ...originArgs ];
```

Or you meant the local variable should being `ReadonlyArray` type?

Reply:
It would still be mutating and creating numerous copies that don't need to be created.  Don't mutate the input argument and there's no need to make copies at all.